sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,673 @@
+        this.clientOptions = clientOptions;
+    }
+    
+    TextAnalyticsAsyncClient(TextAnalyticsClientImpl service, TextAnalyticsServiceVersion serviceVersion) {
+        this(service, serviceVersion, null);
+    }
+
+    /**
+     * Gets the service version the client is using.
+     *
+     * @return the service version the client is using.
+     */
+    public TextAnalyticsServiceVersion getServiceVersion() {
+        return serviceVersion;
+    }
+
+    /**
+     * Returns the detected language and a numeric score between zero and one. Scores close to one indicate 100%
+     * certainty that the identified language is true.
+     *
+     * @param text The text to be analyzed.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} has
+     * the {@link DetectLanguageResult detected language} of the text.
+     * @throws NullPointerException if {@code text} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DetectLanguageResult> detectLanguage(String text) {
+        return detectLanguageWithResponse(text, null).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Returns a {@link Response} containing the detected language and a numeric score between zero and one.
+     * Scores close to one indicate 100% certainty that the identified language is true.
+     *
+     * @param text The text to be analyzed.
+     * @param countryHint Accepts two letter country codes specified by ISO 3166-1 alpha-2. Defaults to "US" if not
+     * specified.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} has the
+     * {@link DetectLanguageResult detected language} of the text.
+     * @throws NullPointerException if {@code text} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DetectLanguageResult>> detectLanguageWithResponse(String text, String countryHint) {
+        try {
+            return withContext(context -> detectLanguageWithResponse(text, countryHint, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<DetectLanguageResult>> detectLanguageWithResponse(String text, String countryHint, Context context) {
+        List<DetectLanguageInput> languageInputs = new ArrayList<>();
+        languageInputs.add(new DetectLanguageInput(Integer.toString(0), text, countryHint));
+        // TODO (savaity):should this be a random number generator?
+        return detectBatchLanguagesWithResponse(languageInputs, null, context).flatMap(response -> {
+            Iterator<DetectLanguageResult> responseItem = response.getValue().iterator();
+            if (responseItem.hasNext()) {
+                return Mono.just(new SimpleResponse<>(response, responseItem.next()));
+            }
+            return monoError(logger, new RuntimeException("Unable to retrieve language for the provided text."));
```

Comment:
Is this what we expect from the service contract? Does the service have an error type (4xx or something) they return if they can't get the language?

Reply:
I think the service still returns a [DocumentError] () with an "Unknown" language and not 40x type errors.

Reply:
RuntimeException is pretty general, should it be an AzureException?